person.rb

```diff
@@ -2,17 +2,30 @@
 require_relative 'decorator'
 require_relative 'capitalize_decorator'
 require_relative 'trimmer_decorator'
-
-# Class representing a person
-class Person
-  attr_accessor :name, :age
-  attr_reader :id
+require_relative 'classroom'
+require_relative 'rental'
+c
+# This Class represent a person
+class Person < Nameable
+  attr_accessor :name, :age, :rentals
+  attr_reader :id, :classroom
 
   def initialize(name, age = 'Unknown', parent_permission: true)
+    super()
     @id = rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def classroom=(classroom)
+    @classroom = classroom
+    classroom.add_student(self)
+  end
```

Comment:
- Hey, this method has been established in the classroom, it is not needed here in the person class. kindly remove it.

Reply:
Done, thank you for the review